### PR TITLE
add figures for id generation

### PR DESCRIPTION
**Pull request by Victor-M-Giraldo**
- added class and sequence diagram for monolith id generation
- rendered new figures
- added figures to relevant section

### CODE REVIEW

**Comment by Victor-M-Giraldo** [comment]
I think the class diagram is relevant. As for the sequence diagram, it's pretty bare bones but I couldn't think of anything else to add because it''s a pretty simple interaction.

File: protocol.typ [hunk 1]
```diff
@@ -84,6 +84,16 @@ These messages conform to the the protocol defined by the #link("https://github.
...
+When a Monolith starts up, and load balancing is enabled, it must listen on a seperate port for incoming balancer connections. Balancers will initiate connections based on their configured Monolith discovery method. Upon accepting a new connection, the Monolith must send an "init" message to the Balancer to inform it of the port that it is listening for normal HTTP requests on, an auth token to verify authenticity, and a MonolithID to identify specific Monolith instances.
 
-Once the Balancer receives this message, the connection is considered fully established, and the Monolith and Balancer can begin sending messages to each other.
\ No newline at end of file
+Once the Balancer receives this message, the connection is considered fully established, and the Monolith and Balancer can begin sending messages to each other.
+
+#figure(
+  image("figures/monolith-id-generation-class.svg", width: 80%),
```
Comment (dyc3) [inline protocol.typ:92]: This needs a better name -- its not just about the ID generation.

File: figures/monolith-id-sequence.mmd [hunk 2]
```diff
@@ -0,0 +1,9 @@
+sequenceDiagram
+
+    participant Monolith
+    participant Balancer
+
+    Note over Monolith: Generates ID
+    Monolith->>Balancer: Sends Init Message with Monolith ID
+    Note over Balancer: Uses ID to initialize connection
+    Balancer->>Monolith: Confirms connection
```
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:9]: There is no confirmation response

File: figures/monolith-id-sequence.mmd
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:1]: perhaps an activity diagram would be a better choice here, similar to `client-connection-state`

File: figures/monolith-id-generation-class.mmd [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    class BalancerConnection {
+        <<abstract>>
+        - id: string
+        - bus: EventEmitter
+        + send(message: MsgM2B): Result<void, Error>
+        + disconnect(): Result<void, Error>
+        - emit(event: BalancerConnectionEvents, ...args: any[]): void
+        + on(event: BalancerConnectionEvents, handler: BalancerConnectionEventHandlers): void
```
Comment (dyc3) [inline figures/monolith-id-generation-class.mmd:22]: feel free to abbreviate some of these function signatures to make the diagram readable

**Comment by Victor-M-Giraldo** [comment]
Changing from sequence to activity diagram now.

File: figures/monolith-id-sequence.mmd
Comment (dyc3) [inline figures/monolith-id-sequence.mmd:1]: remove this since its not being used

**Review by dyc3 — CHANGES_REQUESTED** [review]

**Review by dyc3 — APPROVED** [review]

**Pull request merged**